toter.js

```diff
@@ -141,145 +155,166 @@ switch(command) {
             fs.writeFileSync(settingsPath, JSON.stringify(settings, null, 4), {flag: 'w'})
             rl.close()
         })
-        break;
-    case 'setup':
-        if(!Object.keys(config).length) {
-            let name
-            let description
-            async.series([
-                (callback) => {
-                    rl.question('App/Widget name: ', function(input) {
-                        name = input
-                        callback()
-                    })
-                }, (callback) => {
-                    rl.question('App/Widget description: ', function(input) {
-                        description = input
-                        callback()
-                    })
-                }
-            ], () => {
-                widgetDefaults = {
-                    title: name,
-                    description: description,
-                    minx: 4,
-                    miny: 4,
-                    sizex: 7,
-                    sizey: 7,
-                    icon: 'html'
-                }
-
-                // TODO: Deal with errors nicely
-                config.app_json = JSON.parse(curlHelper('/api/apps', {
-                    name: name,
-                    description: description
-                }))
-
-                stripFields(config.app_json)
-
-                // TODO: Deal with errors nicely
-                config.widget_json = JSON.parse(curlHelper('/api/apps/widgets', Object.assign({}, widgetDefaults, config.widget_json, {
-                    app_id: config.app_json.id,
-                    type: 'marketplace',
-                    source: 'test'
-                })))
-
-                stripFields(config.widget_json)
-                const widgetid = config.widget_json.id
-                delete config.widget_json.id
-
-                curlHelper('/api/storage/buckets/' + config.widget_json.id, {
-                    type: 'public'
-                }, 'put')
-
-                config.widget_json = JSON.parse(curlHelper('/api/apps/widgets/' + widgetid, Object.assign({}, widgetDefaults, config.widget_json, {
-                    type: 'marketplace',
-                    source: '/cmp/api/storage/buckets/' + widgetid + '/' + entry
-                }), 'put'))
-                config.widget_json.id = widgetid
-
-                stripFields(config.widget_json)
-
-                fs.writeFileSync('config.json', JSON.stringify(config, null, 4))
-                rl.close()
-            })
-        } else {
-            console.log('Cannot setup - config contains settings already')
-        }
-        break;
-    case 'update':
-        if(!config.widget_json || config.widget_json && !config.widget_json.id) {
-            console.log('Please run setup first - widget has no ID')
-        } else {
-            let curlObj = Object.assign({}, widgetDefaults, config.widget_json, {
+    }
+}
+
+function setup() {
+    if(!Object.keys(config).length || !config[selectedRegion]) {
```

Comment:
I wonder we might want to give an upgrade path for widgets/apps that have been created already with toter? If there is a widget_json/app_json should we put it in a default object for them?